### PR TITLE
Improved table element styles & fixed table styles inside asides

### LINKED ISSUE #1725: Tables in Asides
### What version of `starlight` are you using?

v0.21.4

### What version of `astro` are you using?

4.5.16

### What package manager are you using?

bun

### What operating system are you using?

Windows

### What browser are you using?

Edge

### Describe the Bug

The background of table is not harmonious when I add it to an aside component.

light mode:
image

dark mode:
image

The source code:

```md
:::note

|     | Low       | Medium     | High       |
| --- | --------- | ---------- | ---------- |
| AFI | 25-30 cm  | 30.1-35 cm | > 35 cm    |
| SDP | 8-11.9 cm | 12-15.9 cm | &ge; 16 cm |

:::
```

### Link to Minimal Reproducible Example

none

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by SnowDingo**
This PR brings two changes to Astro Starlight:

1.  It closes #1725, which addresses the broken style in the table inside the aside element.
2.  It also adds new styling to the table element in Astro, with most of this part of the code from @delucis.

Here are sample images of the new styling:
スクリーンショット 2024-06-29 9 30 42
スクリーンショット 2024-06-29 9 30 47
スクリーンショット 2024-06-29 9 31 13
スクリーンショット 2024-06-29 9 31 23
スクリーンショット 2024-06-29 9 32 16
スクリーンショット 2024-06-29 9 34 06
スクリーンショット 2024-06-29 9 35 08
スクリーンショット 2024-06-29 9 35 14
スクリーンショット 2024-06-29 9 35 22
スクリーンショット 2024-06-29 9 35 29
スクリーンショット 2024-06-29 9 35 37
スクリーンショット 2024-06-29 9 35 48

### CODE REVIEW

**Review by delucis — APPROVED** [review]
Thank you @SnowDingo! I made some changes as I think there were some conflicts between the styles you applied and other recent CSS changes to other bits of Starlight and I removed the background striping to match my example StackBlitz.

I’ve temporarily added a test page to the Starlight docs on this branch so people can view the result for a range of examples and check it in both light and dark mode: https://starlight-git-fork-snowdingo-main-astrodotbuild.vercel.app/guides/tables-test/

I’m personally happy with this, but would love to hear feedback from anyone else who has thoughts!

**Comment by lorenzolewis** [comment]
For me it seems like the blue border is a bit too red and in the purple-ish territory instead of blue. 

image

image

File: packages/starlight/style/markdown.css
Comment (HiDeoo) [inline packages/starlight/style/markdown.css:134]:
```suggestion
@supports (border-color: color-mix(in srgb, var(--sl-color-asides-text-accent) 30%, transparent)) {
	.sl-markdown-content .starlight-aside td:not(:where(.not-content *)) {
		border-color: color-mix(in srgb, var(--sl-color-asides-text-accent) 30%, transparent);
	}
}
```

Sadly, it looks like we would need to change the color space to avoid the issue mentioned by @lorenzolewis which is this bug: https://bugs.webkit.org/show_bug.cgi?id=268463

**Review by sarah11918 — APPROVED** [review]
Looks great! Some really nice styling here!

**Comment by evadecker** [comment]
Agree with @lorenzolewis about the blue dividers having a purple tint.

The header row divider feels too strong; I would lower the opacity or match the tint of the header text for the aside.

The final divider of the table is subtly lighter than all the other dividers.

image

**Comment by delucis** [comment]
Thanks for the reviews folks — super helpful! I will play around with the header border and inline padding @evadecker 🙌 

> The final divider of the table is subtly lighter than all the other dividers.

@evadecker Could you share a bit more about the browsers/hardware you’re using? I’ve checked a bunch of different browsers on macOS and can’t see the same thing. Wondering if its a subpixel antialiasing effect at a specific resolution 🤔

**Comment by evadecker** [comment]
> Thanks for the reviews folks — super helpful! I will play around with the header border and inline padding @evadecker 🙌
> 
> > The final divider of the table is subtly lighter than all the other dividers.
> 
> @evadecker Could you share a bit more about the browsers/hardware you’re using? I’ve checked a bunch of different browsers on macOS and can’t see the same thing. Wondering if its a subpixel antialiasing effect at a specific resolution 🤔

First noticed on iOS Safari and the issue is present on macOS Safari as well. Not present on Chrome.

The issue seems to stem from [`border-collapse: collapse`](https://github.com/withastro/starlight/pull/2064/files#diff-8fb3b63b811f8b141bfcc05628c305d07580fe282ef850de3e0ff2414a89ec79L106). 

Removing that rule and adding `border-spacing: 0` fixes the issue.

https://github.com/withastro/starlight/assets/4117920/380387fd-ae7b-4800-98e6-03fa5c9636ed

File: packages/starlight/style/markdown.css
Comment (evadecker) [inline packages/starlight/style/markdown.css:107]:
```suggestion
	border-spacing: 0;
```

Comment (delucis) [reply packages/starlight/style/markdown.css:107]: Ah, great catch!

**Comment by delucis** [comment]
OK, made two more changes based on @evadecker’s feedback:

1. Removed the inline padding at the start and end of each row.
2. Simplified the colours by using the same colour for the heading row divider as the rest of the rows.

| Before | After |
|---|---|
| starlight-git-fork-snowdingo-main-astrodotbuild vercel app_guides_tables-test_ | localhost_4321_guides_tables-test_ |

**Review by evadecker — APPROVED** [review]
Looks great! Excited to see this go live!

**Review by lorenzolewis — APPROVED** [review]
Yes, looks perfect now for me in the always-problematic browser that is Safari 😉

**Comment by SnowDingo** [comment]
So is this PR ready to go?
If not, can someone make a list of changes I have to make before it gets merged to the main Starlight branch?


> Yes, looks perfect now for me in the always-problematic browser that is Safari 😉

Does anyone know if this PR is ready to go?
If not, since I couldn't catch up to the comments, it would be conducive if someone could create a list of changes we must make before merging this PR.

**Comment by libreom** [comment]
Screenshot_20240630-072856_Brave
Screenshot_20240630-072907_Brave
Not so good

**Comment by delucis** [comment]
> Does anyone know if this PR is ready to go?

@SnowDingo Yes! Good to go. We will merge it when we are ready for the next release. Should be later this week. Thanks again!

**Comment by SnowDingo** [comment]
> > Does anyone know if this PR is ready to go?
> 
> @SnowDingo Yes! Good to go. We will merge it when we are ready for the next release. Should be later this week. Thanks again!

@delucis , no problem👍
I can't wait to see our new styling in starlight!!

**Pull request merged**
